tests/mock_network/churn.rs

```diff
@@ -314,36 +324,22 @@ fn drop_random_nodes<R: Rng>(
         }
 
         let elder_size = unwrap!(node.inner.elder_size());
+        let safe_section_size = unwrap!(node.inner.safe_section_size());
+
         let section = unwrap!(sections.get_mut(node.our_prefix()));
 
-        // Don't drop any other node if an elder is already scheduled for drop.
-        if section.dropped_elder_count > 0 {
+        // Drop at most as many nodes as is the minimal number of non-elders in the section.
+        // This guarantees we never drop below `elder_size` even in case of split.
+        if section.dropped_count >= safe_section_size - elder_size {
             continue;
         }
 
         // Don't drop below elder_size nodes.
-        if section.all_remaining() <= elder_size {
+        if section.initial_count <= elder_size {
             continue;
         }
 
-        // If there already are other drops scheduled, make sure we remain with at least one
-        // more node above elder_size. This is because one of those other drops might trigger
-        // relocation of one of the existing elders and we wouldn't have anyone to replace it with
-        // otherwise.
-        if section.all_dropped() > 0 && section.all_remaining() <= elder_size + 1 {
-            continue;
-        }
-
-        if node.inner.is_elder() {
-            // Don't drop elder if a non-elder is already scheduled for drop.
```

Comment:
so this situation is no longer an issue now?

Reply:
I'd say so. I think we should be able to drop any number of elders as long as we have enough non-elders to replace them with.